default/python/AI/AIstate.py

```diff
@@ -382,53 +381,64 @@ def update_system_status(self):
-                        ship_stats = CombatRatingsAI.FleetCombatStats(fleet_id).get_ship_stats(hashable=True)
-                        # track old/dead enemy fighters for rating assessments in case not enough current info
-                        e_f_dict = old_e_fighters if dead_fleet else cur_e_fighters
-                        for stats in ship_stats:
-                            attacks = stats[0]
-                            if attacks:
-                                e_f_dict.setdefault(stats, [0])[0] += 1
-                    visibility_turns_map = universe.getVisibilityTurnsMap(fleet_id, empire_id)
-                    partial_vis_turn = visibility_turns_map.get(fo.visibility.partial, -9999)
-                    if not dead_fleet:
-                        # TODO: consider checking death of individual ships.  If ships had been moved from this fleet
-                        # into another fleet, we might have witnessed their death in that other fleet but if this fleet
-                        # had not been seen since before that transfer then the ships might also still be listed here.
-                        sys_status = self.systemStatus.setdefault(this_system_id, {})
-                        sys_status['enemy_ship_count'] = sys_status.get('enemy_ship_count', 0) + len(fleet.shipIDs)
-                        if partial_vis_turn >= current_turn - 1:  # only interested in immediately recent data
-                            saw_enemies_at_system[fleet.systemID] = True
-                            enemy_fleet_ids.append(fleet_id)
-                            enemies_by_system.setdefault(this_system_id, []).append(fleet_id)
-                            if not fleet.ownedBy(-1):
-                                self.misc.setdefault('enemies_sighted', {}
-                                                     ).setdefault(current_turn, []).append(fleet_id)
-                                rating = CombatRatingsAI.get_fleet_rating(
-                                    fleet_id, enemy_stats=CombatRatingsAI.get_empire_standard_fighter())
-                                if rating > 0.25 * my_milship_rating:
-                                    self.misc.setdefault('dangerous_enemies_sighted', {}
-                                                         ).setdefault(current_turn, []).append(fleet_id)
+            # TODO: check if currently in system and blockaded before accepting destination as location
+            this_system_id = fleet.nextSystemID if fleet.nextSystemID != INVALID_ID else fleet.systemID
+            dead_fleet = fleet_id in destroyed_object_ids
+
+            if fleet.ownedBy(empire_id):
+                if not dead_fleet:
+                    my_fleets_by_system.setdefault(this_system_id, []).append(fleet_id)
+                    fleet_spot_position.setdefault(fleet.systemID, []).append(fleet_id)
+                continue
+
+            # this is a fleet not owned by us
```

Comment:
`military fleet`

Reply:
May be totally remove this comment?

Reply:
The comment seems fine to me as-is.   Sure, it just explicitly states something that could be inferred from reviewing a few lines of code before it, but that's the most typical point of comments, is it not, to make some aspect of the code particularly clear.  And it seems to me it would not be accurate to call it a military fleet in this comment.

Reply:
Ok, I am confused by this comment.

Reply:
I understand that Cj interprets the comment to refer to the lines below while the comment is meant to clarify the situation after the previous code.

I will make the comment more clear after merging the subsequent PRs to avoid annoying conflicts.